packages/contractkit/src/wrappers/DoubleSigningSlasher.ts

```diff
@@ -0,0 +1,73 @@
+import { Address } from '../base'
+import { DoubleSigningSlasher } from '../generated/types/DoubleSigningSlasher'
+import {
+  BaseWrapper,
+  CeloTransactionObject,
+  proxyCall,
+  toTransactionObject,
+  valueToInt,
+} from './BaseWrapper'
+
+/**
+ * Contract handling slashing for Validator double-signing
+ */
+export class DoubleSigningSlasherWrapper extends BaseWrapper<DoubleSigningSlasher> {
+  getEpochNumberOfBlock = proxyCall(
+    this.contract.methods.getEpochNumberOfBlock,
+    undefined,
+    valueToInt
+  )
+
+  getBlockNumberFromHeader = proxyCall(
+    this.contract.methods.getBlockNumberFromHeader,
+    undefined,
+    valueToInt
+  )
+
+  async slash(
+    signerIndex: number,
+    headerA: string,
+    headerB: string
+  ): Promise<CeloTransactionObject<void>> {
+    const blockNumber = await this.getBlockNumberFromHeader([headerA])
+    const blockEpoch = await this.getEpochNumberOfBlock(blockNumber)
+
+    const validators = await this.kit.contracts.getValidators()
+    const signer: Address = await this.contract.methods
+      .validatorSignerAddressFromSet(signerIndex, blockNumber)
+      .call()
+
+    const validator = await validators.getValidatorFromSigner(signer)
+    const history = await validators.getValidatorMembershipHistory(validator.address)
+    const historyIndex = validators.findMembershipHistoryIndexForEpoch(history, blockEpoch)
+    const group = history[historyIndex].group
+
+    const election = await this.kit.contracts.getElection()
+    const eligibleGroups = await election.getEligibleValidatorGroupsVotes()
+    const validatorVotes = await election.findLessersAndGreaters(
+      await election.getGroupsVotedForByAccount(validator.address),
+      eligibleGroups
+    )
+    const groupVotes = await election.findLessersAndGreaters(
+      await election.getGroupsVotedForByAccount(group),
+      eligibleGroups
+    )
+
+    return toTransactionObject(
```

Comment:
we might be able to do a better job on the data we send. Since we won't be removing all votes, only the necessary ones to complete the penalty, right?
In that case, we could send less lesser/greater arrays..

Reply:
Updated to use Sami's utilities, which only send the necessary deltas.